### PR TITLE
Convert petsc tests 1

### PR DESCRIPTION
**Pull request by drwells**
This PR converts our tests for `PETScWrappers::Vector` to use `PETScWrappers::MPI::Vector` and also `PETScWrappers::BlockVector` to `PETScWrappers::MPI::BlockVector`. As discussed in #4279 this requires removing the class `PETScWrappers::BlockSparseMatrix` (though we can keep `PETScWrappers::SparseMatrix`).

This looks like a lot, but I used a few macros to do most of the work. I tried to put similar changes into the same commits to make things easier to read.

### CODE REVIEW

**Comment by drwells** [comment]
*ping*

File: tests/petsc/vector_equality_1.cc [hunk 1]
```diff
@@ -15,18 +15,18 @@
 
 
 
-// check PETScWrappers::Vector::operator==(PETScWrappers::Vector) for vectors that are not
+// check PETScWrappers::MPI::Vector::operator==(PETScWrappers::MPI::Vector) for vectors that are not
 // equal
```
Comment (masterleinad) [inline tests/petsc/vector_equality_1.cc]: If you are already reformatting at other places, may be do so here as well?

File: tests/petsc/vector_equality_4.cc [hunk 2]
```diff
@@ -15,18 +15,18 @@
 
 
 
-// check PETScWrappers::Vector::operator!=(PETScWrappers::Vector) for vectors that are
+// check PETScWrappers::MPI::Vector::operator!=(PETScWrappers::MPI::Vector) for vectors that are
 // equal
```
Comment (masterleinad) [inline tests/petsc/vector_equality_4.cc]: reformat here as well?

**Comment by masterleinad** [comment]
I didn't look at the commits individually. What about squashing?

**Comment by drwells** [comment]
I adjusted the lines and rebased.

**Pull request merged**